firmware/src/core/wifiWidget.cpp

```diff
@@ -96,27 +101,29 @@ void WifiWidget::update(bool force) {
 
 void WifiWidget::draw(bool force) {
     // Force is currently unhandled due to not knowing what behavior it would change
-    TFT_eSPI &display = m_manager.getDisplay();
     m_manager.selectScreen(statusScreenIndex);
     const int blankRectTop = ScreenCenterY + lineHeight / 2;
 
     if (!m_isConnected && !m_connectionFailed) {
-        display.fillRect(0, blankRectTop, ScreenWidth, ScreenHeight - blankRectTop, TFT_BLACK);
-        display.drawCentreString(m_dotsString, ScreenCenterX, ScreenCenterY + lineHeight, 4);
+        m_manager.fillRect(0, blankRectTop, ScreenWidth, ScreenHeight - blankRectTop, TFT_BLACK);
+        m_manager.drawCentreString(m_dotsString.c_str(), ScreenCenterX, ScreenCenterY + lineHeight, fontSize);
     } else if (m_isConnected && !m_hasDisplayedSuccess) {
         m_hasDisplayedSuccess = true;
-        display.fillScreen(TFT_BLACK);
-        display.drawCentreString("Success", ScreenCenterX, ScreenCenterY, 4);
+        m_manager.clearScreen();
+        m_manager.drawCentreString("Success", ScreenCenterX, ScreenCenterY, fontSize);
         m_manager.selectScreen(statusScreenIndex + 1);
-        display.drawCentreString("IP Address", ScreenCenterX, ScreenCenterY - lineHeight, 4);
-        display.drawCentreString(m_ipaddress, ScreenCenterX, ScreenCenterY + lineHeight, 4);
+        m_manager.clearScreen();
+        m_manager.drawCentreString("IP Address", ScreenCenterX, ScreenCenterY - lineHeight, fontSize);
+        m_manager.drawCentreString(m_ipaddress.c_str(), ScreenCenterX, ScreenCenterY + lineHeight, fontSize);
         Serial.println();
         Serial.println("Connected to WiFi");
         m_isConnected = true;
+        delay(messageDelay);
     } else if (m_connectionFailed && !m_hasDisplayedError) {
         m_hasDisplayedError = true;
-        display.fillRect(0, blankRectTop, ScreenWidth, ScreenHeight - blankRectTop, TFT_BLACK);
-        display.drawCentreString(m_connectionString, ScreenCenterX, ScreenCenterY + lineHeight, 4);
+        m_manager.fillRect(0, blankRectTop, ScreenWidth, ScreenHeight - blankRectTop, TFT_BLACK);
+        m_manager.drawCentreString(m_connectionString.c_str(), ScreenCenterX, ScreenCenterY + lineHeight, fontSize);
+        delay(messageDelay);
```

Comment:
Why are those delays here?
That seems unrelated to drawing.

Reply:
Without the delays, the screens are immediately cleared and you can't see the message.